pkg/client/client.go

```diff
@@ -0,0 +1,121 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/weaveworks/footloose/pkg/api"
+	"github.com/weaveworks/footloose/pkg/config"
+)
+
+// Client is a object able to talk a remote footloose API server.
+type Client struct {
+	baseURI string
+	client  *http.Client
+}
+
+// New creates a new Client.
+func New(baseURI string) *Client {
+	return &Client{
+		baseURI: baseURI,
+		client:  &http.Client{},
+	}
+}
+
+func (c *Client) uriFromPath(path string) string {
+	return c.baseURI + path
```

Comment:
This is sensitive to URL syntax details (e.g. a trailing slash at the end of baseURI makes a difference).
To make URL manipulation safer, we might change the type of `c.baseURI` to type `net/purl.URL` (instead of `string`), and replace this line with something like `c.baseURI.ResolveReference(path)`.